### PR TITLE
Update egress selector and namespace in e2e

### PR DESCRIPTION
**Pull request by nshankar13**
**Please provide a description of this PR:**

Egress gateway e2es still had a few remaining tests that assumed / hardcoded the namespace as `istio-system` and label / service name as `egressgateway` and `istio-egressgateway`.

Resolves https://github.com/istio/istio/issues/45663

### CODE REVIEW

File: tests/integration/pilot/tunneling_test.go [hunk 1]
```diff
@@ -120,15 +120,37 @@ func TestTunnelingOutboundTraffic(t *testing.T) {
 			externalForwardProxyIP := getPodIP(ctx, externalNs, "external-forward-proxy")
 
 			for _, proxyConfig := range forwardProxyConfigurations {
+				var egressNs string
+				var egressSvc string
+				var egressLabel string
+
+				if i.Settings().EgressGatewayServiceNamespace != "" {
```
Comment (howardjohn) [inline tests/integration/pilot/tunneling_test.go:127]: tests should not have to deal with this complexity (and won't - its guaranteed  someone will use `i.Settings().EgressGatewayServiceNamespace` on its own and break things silently). Just make `i.Settings().EgressGatewayServiceNamespace` default to `istio-system`

Comment (nshankar13) [reply tests/integration/pilot/tunneling_test.go:127]: same for the egress svc name and label too?

Comment (howardjohn) [reply tests/integration/pilot/tunneling_test.go:127]: yes

Comment (nshankar13) [reply tests/integration/pilot/tunneling_test.go:127]: The ingress and egress vars already say that they default to `istio-system` and the default labels, but I'm not sure where those defaults are set. https://github.com/istio/istio/blob/master/pkg/test/framework/components/istio/config.go#L171-L185

Would we need to set them here? https://github.com/istio/istio/blob/master/pkg/test/framework/components/istio/config.go#L78-L89

Comment (nshankar13) [reply tests/integration/pilot/tunneling_test.go:127]: Also, some tests (ex: https://github.com/istio/istio/blob/master/tests/integration/security/filebased_tls_origination/egress_gateway_origination_test.go#L235) don't have `istio-egressgateway` as the default istio label.

Comment (nshankar13) [reply tests/integration/pilot/tunneling_test.go:127]: Nvm, I guess the default label is usually `egressgateway`, whereas the default Service name is `istio-egressgateway`?

File: tests/integration/security/filebased_tls_origination/egress_gateway_origination_test.go [hunk 2]
```diff
@@ -299,10 +301,18 @@ spec:
...
-func createGateway(t test.Failer, ctx resource.Context, appsNamespace namespace.Instance, serviceNamespace namespace.Instance) {
+func createGateway(t test.Failer, ctx resource.Context, appsNamespace namespace.Instance,
+	serviceNamespace namespace.Instance, egressNs string, egressSvc string, egressLabel string,
+) {
 	ctx.ConfigIstio().
-		Eval(appsNamespace.Name(), map[string]string{"ServerNamespace": serviceNamespace.Name()}, Gateway).
-		Eval(appsNamespace.Name(), map[string]string{"ServerNamespace": serviceNamespace.Name()}, VirtualService).
+		Eval(appsNamespace.Name(), map[string]string{
+			"ServerNamespace": serviceNamespace.Name(),
+			"EgressNamespace": egressNs, "EgressLabel": egressLabel, "EgressService": egressSvc,
```
Comment (keithmattix) [inline tests/integration/security/filebased_tls_origination/egress_gateway_origination_test.go:310]: nit: was gofmt rub here? This looks off

**Review by howardjohn — APPROVED** [review]

File: tests/integration/security/filebased_tls_origination/egress_gateway_origination_test.go (hunk 2 above)
Comment (nshankar13) [reply tests/integration/security/filebased_tls_origination/egress_gateway_origination_test.go:310]: Yeah, this was after running `gofmt` and `gofumpt`.

**Pull request merged**